src/term-spec.js

```diff
@@ -128,21 +128,135 @@ declare export class AssignmentTargetPropertyProperty extends AssignmentTargetPr
 // class
 declare export class ClassExpression extends Expression {
   name?: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassDeclaration extends Statement {
+  isAbstract: any; // boolean
   name: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassElement extends Term {
-  isStatic: any;
+  isStatic: any; // boolean
+  accessModifier: any; // 'public' | 'protected' | 'private'
+}
+
+declare export class MethodClassElement extends ClassElement {
+  isAbstract: any; // boolean
+  hasQuestionToken: any; // boolean
   method: MethodDefinition;
 }
 
+declare export class ConstructorClassElement extends ClassElement {
+  params: ConstructorFormalParameters;
+  body: FunctionBody;
+}
+
+declare export class ConstructorFormalParameters extends Term {
+  items: ConstructorParameterDeclaration[];
+  rest?: ParameterDeclaration;
+}
+
+declare export class ConstructorParameterDeclaration extends Term {
+  accessModifier?: any; // 'public' | 'protected' | 'private'
+  hasReadonlyModifier: any; // boolean
+  parameter: ParameterDeclaration;
+}
+
+declare export class SemicolonClassElement extends ClassElement {}
```

Comment:
This sneaked in from [the TypeScript source](https://github.com/Microsoft/TypeScript/blob/4c824505ade0740cf2848ebd862eccbabd1771db/src/compiler/types.ts). The comment in there says the following:

> For when we encounter a semicolon in a class declaration. ES6 allows these as class elements.

So it's perhaps only used for recording and later emitting these empty semicolons [as can be seen here](https://www.typescriptlang.org/play/#src=class C {%0D%0A %3B%3B%3B%0D%0A}).

When I added it, I probably thought it had some deep significance. Shall I remove it?